pkg/cache/context.go

```diff
@@ -1294,41 +1295,68 @@ func (ctx *Context) HandleContainerStateUpdate(request *si.UpdateContainerSchedu
 func (ctx *Context) ApplicationEventHandler() func(obj interface{}) {
 	return func(obj interface{}) {
 		if event, ok := obj.(events.ApplicationEvent); ok {
-			app := ctx.GetApplication(event.GetApplicationID())
+			appID := event.GetApplicationID()
+			app := ctx.GetApplication(appID)
 			if app == nil {
-				log.Log(log.ShimContext).Error("failed to handle application event",
-					zap.String("reason", "application not exist"))
+				log.Log(log.ShimContext).Error("failed to handle application event, application does not exist",
+					zap.String("applicationID", appID))
 				return
 			}
+
 			if app.canHandle(event) {
 				if err := app.handle(event); err != nil {
 					log.Log(log.ShimContext).Error("failed to handle application event",
 						zap.String("event", event.GetEvent()),
+						zap.String("applicationID", appID),
 						zap.Error(err))
+					return
 				}
 			}
+			log.Log(log.ShimContext).Error("application event cannot be handled in the current state",
```

Comment:
This log message is weird to me since the application could be get handled successfully rather than "cannot be handled" in this path. Maybe we need to add return to `if app.canHandle(event) {`

Reply:
Correct, I modified that. Initially it was a bunch of if-else which was hard to read, I do hope this is better.